### DISCUSSION TITLE
slidev as a module inside an existing project

### DISCUSSION THREAD

**Discussion by nort3x** [slidevjs/slidev#1836] [Q&A]
i need to create different instances of slidev as part of a nuxt project,
i can use nginx to redirect the request to relevant module but i want to start by demanding slidev act as a single all-in-one component in my vue project,
there are many opinionated choices made (like UnoCSS, having a cli and ...) which is an awesome bundle for who wants to create a slide, but i require some more degree of flexibility here,
please let me know if there is any hope here,

it would be awesome if i could:

```vue
<slidev-main :pages="pages" :config="config"/>
```
and have embedded view however i need

what do you think?

**Comment by kermanx** [comment]
This is planned, but may not be implemented soon, because this requires rewriting lots of things. 🤔

Currently `<iframe>` is the only way to embed the slides.

**Reply by nort3x** [reply]
that's great to hear about,
i'm planning to use this for a non-profit educational program and managing this with `iframe` is beyond my abilities,
optimal solution would be bind one huge json and you are good to go,
i can offer my help but probably one of the core members would be much more efficient in making it work but you have my keyboard if required 
- salute
